File: RETO383.java
//Rueskas
import java.util.*;

public class RETO383
{
    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);

        int cases, pages,min, reach;
        int[] jumps;

        cases = sc.nextInt();
        sc.nextLine();

        for (int c = 0; c < cases; c++)
        {
            pages = sc.nextInt();
            jumps = new int[pages];
            for (int i = 0; i < pages; i++)
            {
                jumps[i] = sc.nextInt();
            }
            min = jumps[pages - 1];
            reach = jumps[pages - 2] - jumps[pages - 1];
            
            for (int i = pages-2; i >= 0; i--)
            {
                if(jumps[i] < min)
                    min = jumps[i];
                else if (jumps[i] - min > reach)
                    reach = jumps[i] - min;
            }
            System.out.println(reach);
        }
    }
}

File: RETO383.cpp
//Rueskas
//Pos 10
//0.0092 Sec
#include <iostream>
using namespace std;

int main()
{
    int cases, pages, min, reach;
    scanf("%d", &cases);
    for (int i = 0; i < cases; i++)
    {
        scanf("%d", &pages);
        int jumps[pages];
        for (int j = 0; j < pages; j++)
        {
            scanf("%d", &jumps[j]);
        }
        
        min = jumps[pages - 1];
        reach = jumps[pages-2] - jumps[pages - 1];
        
        for (int z = pages-2; z >= 0 ; z--)
        {
            if(jumps[z] < min)
                min = jumps[z];
            else if (jumps[z] - min > reach)
                reach = jumps[z] - min;
        }
        
        printf("%d\n", reach);
    }
    return 0;
}
